lib/queryTicket.js

```diff
@@ -1,61 +1,43 @@
-import fetch from 'node-fetch'
-import { parseString } from 'xml2js'
 import { parseLegs, price } from './parse.js'
-
-const endpoint = 'https://fahrkarten.bahn.de/mobile/dbc/xs.go?'
+import requestBackend from './requestBackend.js';
 
 /**
  * queryTicket function
  *
  * @param   string | DB ticket number (Auftragsnummer)
  * @param   lastName | last name of purchaser 
+ * @param   kwid | kwid of the ticket (only needed for tickets from the new system)
  * @return  Promise | fulfills with an object that contains the ticket data
  */
-const queryTicket = async (ticketNumber, lastName) => {
-
-  const reqBody = '<?xml version="1.0"?><rqorderdetails version="1.0">'
-    + '<rqheader tnr="61743782011" ts="2022-07-15T22:29:00" l="de" v="22060000" d="iPhone13,1" os="iOS_15.5" app="NAVIGATOR"/>'
-    + '<rqorder on="' + ticketNumber + '"/><authname tln="' + lastName + '"/></rqorderdetails>'
+const queryTicket = async (ticketNumber, lastName,kwid=null) => {
 
-  return fetch(endpoint, {
-    method: 'POST',
-    headers: {
-      'User-Agent': 'github.com/envake/db-tickets',
-      'Accept-Language': 'de-DE,de;q=0.9'
-    },
-    body: reqBody
-  })
-  .then(response => {
-    if (!response.ok) {
-      throw new HTTPResponseError(response)
-    }
-    return response.text()
-  })
-  .then(body => {
-    return new Promise((resolve, reject) => parseString(body, (err, result) => {
-      if (err) {
-        reject(err)
-      }
-      else {
-        resolve(result)
-      }
-    }))
-  })
-  .then(json => {
-    if (json['rperror']) {
-      throw new XMLRPCError(json['rperror']['error'])
+  let reqBody = '<?xml version="1.0"?><rqorderdetails version="2.0">'
```

Comment:
I think a template string would make this more readable. For example, we could do it like this:

```js
  let reqBody = `\
    <?xml version="1.0"?>
    <rqorderdetails version="2.0">
      <rqheader l="de" v="23080000" d="iPhone16.4.1" os="iOS_15.7.5" app="NAVIGATOR" />
      <rqorder on="${ticketNumber}" ${kwid ? `kwid="${kwid}" ` : ''}/>
      <authname tln="${lastName}"/>
    </rqorderdetails>
  `
```